src/escape.rs

```diff
@@ -1799,3 +1799,17 @@ fn parse_number(bytes: &str, range: Range<usize>) -> Result<char, EscapeError> {
         None => Err(EscapeError::InvalidCodepoint(code)),
     }
 }
+
+#[inline]
+fn from_str_radix(src: &str, radix: u32) -> Result<u32, ParseIntError> {
+    let codepoint = u32::from_str_radix(src, radix)?;
+
+    // If we here then `src` is not empty, otherwise u32::from_str_radix returns error
+    match src.as_bytes()[0] {
+        // We should not allow sign numbers, but u32::from_str_radix will accept them.
+        // Because we cannot directly construct necessary error, we use the function
+        // which would return it for us
+        b'+' | b'-' => u32::from_str_radix("=", 10),
```

Comment:
I don't like this very much, personally.  I can see the appeal of using `from_str_radix` and `ParseIntError`, the overlap is pretty significant, but this is definitely a hack around the mismatch in semantics.  I'd rather define our own `ParseIntError` (or equivalent) that is easily convertible from the original and can be easily returned on its own.  You can copy the variants and descriptions straight from the original.

Reply:
After thinking I realized that here we needed our own type of error in any case (`ParseCharRefError`), into which I moved the variants that make sense only when parsing the character reference.